x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_list/credentials_list.test.tsx

```diff
@@ -0,0 +1,276 @@
+        expect(wrapper.text()).toEqual('Private API Key');
+      });
+    });
+
+    describe('column 3 (key)', () => {
+      const testToken = {
+        ...apiToken,
+        key: 'abc-123',
+      };
+
+      const subject = (token: object): [any, any] => {
+        const copyMock = jest.fn();
+        const column = columns[2];
+        const wrapper = shallow(<div>{column.render(token)}</div>);
+        const children = wrapper.find(EuiCopy).props().children;
+        const copyEl = shallow(children(copyMock));
+        return [copyMock, copyEl];
+      };
+
+      it('renders the credential and a button to copy it', () => {
+        const [copyMock, copyEl] = subject(testToken);
+        expect(copyEl.find('EuiButtonIcon').props().onClick).toEqual(copyMock);
+        expect(copyEl.text().replace('<EuiButtonIcon />', '')).toBe('abc-123');
+      });
+
+      it('empty text if no key is present', () => {
+        const [copyMock, copyEl] = subject({
+          ...testToken,
+          key: undefined,
+        });
+        expect(copyEl.find('EuiButtonIcon').props().onClick).toEqual(copyMock);
+        expect(copyEl.text().replace('<EuiButtonIcon />', '')).toBe('');
+      });
+    });
+
+    describe('column 4 (modes)', () => {
+      const token = {
+        ...apiToken,
+        type: ApiTokenTypes.Admin,
+      };
+
+      it('renders correctly', () => {
+        const column = columns[3];
+        const wrapper = shallow(<div>{column.render(token)}</div>);
+        expect(wrapper.text()).toEqual('--');
+      });
+    });
+
+    describe('column 5 (engines)', () => {
+      const token = {
+        ...apiToken,
+        type: ApiTokenTypes.Private,
+        access_all_engines: true,
+      };
+
+      it('renders correctly', () => {
+        const column = columns[4];
+        const wrapper = shallow(<div>{column.render(token)}</div>);
+        expect(wrapper.text()).toEqual('all');
+      });
+    });
+
+    describe('column 6 (edit action)', () => {
+      const token = apiToken;
+
+      it('calls showCredentialsForm when clicked', () => {
+        const action = columns[5].actions[0];
+        action.onClick(token);
+        expect(showCredentialsForm).toHaveBeenCalledWith(token);
+      });
+    });
+
+    describe('column 7 (delete action)', () => {
+      const token = {
+        ...apiToken,
+        name: 'some-name',
+      };
+
+      it('calls deleteApiKey when clicked', () => {
+        const action = columns[5].actions[1];
+        action.onClick(token);
+        expect(deleteApiKey).toHaveBeenCalledWith('some-name');
+      });
+    });
+  });
+
+  describe('onChange', () => {
+    it('will handle pagination by calling `fetchCredentials`', () => {
+      const fetchCredentials = jest.fn();
+
+      mockKea({ actions: { fetchCredentials } });
+      const wrapper = shallow(<CredentialsList />);
+      const onChange = wrapper.find(EuiBasicTable).props().onChange;
```

Comment:
destructuring opportunity if we want to be fancy
```suggestion
      const { onChange } = wrapper.find(EuiBasicTable).props();
```